### DISCUSSION TITLE
Update for 10.0.24?

### DISCUSSION THREAD

**Discussion by FH-Inway** [d365collaborative/d365fo.tools#641] [General]
The new 10.0.24 VHD template from Microsoft introduced several changes. Some of those have an impact on the d365fo.tools. Most noteably: It seems that the module cannot be installed without the `-Clobber` option.

I'm proposing to start a collection of changes and potential issues and improvements for the d365fo.tools resulting from them.

Another topic for discussion: Can and should the module support both pre and post 10.0.24 VHD environments? And if yes, how long should the support last?

**Comment by FH-Inway** [comment]
Collection of issues and solutions:
- #646 
  - #657 
- #642 
- #640 
  - #361 
- #633 
- #587 
- https://www.yammer.com/dynamicsaxfeedbackprograms/threads/1682828402622464
- https://www.yammer.com/dynamicsaxfeedbackprograms/#/Threads/show?threadId=1472552208572416

**Comment by Splaxi** [comment]
I'm all in for getting things back on track in terms of making it as easy as possible for people to install and run the tools.

As you correctly point out - what do we do about support for PRE / POST 10.0.24 VHD (Onebox / Local)? The general rule has been that we follow the version enforcement of Microsoft and when possible keep things backwards compatible, unless it requires to much effort.

Take the provisioning tool - there was several changes over the course of time, where we had to do mini hacks in the code, to handle PRE / POST specific PU's.

With that statement, we should strive for getting the tools to install and run flawlessly on any CHE / VHD with 10.0.24 - if they are coming with the same set PowerShell modules installed. It should be fairly easy to identify which ones that are conflicting with our current prereq modules. From there we can plan our work, in terms of what cmdlets are using which modules and create specific tasks that needs to be completed.

**Comment by FH-Inway** [comment]
Sounds like a plan. I was thinking of creating a PowerShell script that would list all components (PowerShell modules, versions of OS, SQL Server, ...) to make it easier to compare the VHDs. Would have to be a stand alone script so that it can be downloaded and executed without installing the d365fo.tools.
